### ISSUE TITLE
TypeError: Cannot read property 'path' of null

### ISSUE THREAD

**Issue by polytechnic** [kriasoft/react-starter-kit#152]
New clone, When i open new blank window and load component pages contact and login from

http://localhost:3000/contact
http://localhost:3000/login

 i get returned:

```
TypeError: Cannot read property 'path' of null
   at Object.ID_1 (/Users/username/Sites/myapp/build/webpack:/src/stores/AppStore.js:69:26)
   at Dispatcher.$Dispatcher_invokeCallback (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:220:35)
   at Dispatcher.dispatch (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:195:14)
   at /Users/username/Sites/myapp/build/webpack:/src/core/Database.js:45:18
   at run (/Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:89:1)
   at /Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:100:1
   at process._tickCallback (node.js:355:11)
```

And on the terminal
```
TypeError: Cannot read property 'path' of null
    at Object.ID_1 (/Users/username/Sites/myapp/build/webpack:/src/stores/AppStore.js:69:26)
    at Dispatcher.$Dispatcher_invokeCallback (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:220:35)
    at Dispatcher.dispatch (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:195:14)
    at /Users/username/Sites/myapp/build/webpack:/src/core/Database.js:45:18
    at run (/Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:89:1)
    at /Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:100:1
    at process._tickCallback (node.js:355:11)
TypeError: Cannot read property 'path' of null
    at Object.ID_1 (/Users/username/Sites/myapp/build/webpack:/src/stores/AppStore.js:69:26)
    at Dispatcher.$Dispatcher_invokeCallback (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:220:35)
    at Dispatcher.dispatch (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:195:14)
    at /Users/username/Sites/myapp/build/webpack:/src/core/Database.js:45:18
    at run (/Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:89:1)
    at /Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:100:1
    at process._tickCallback (node.js:355:11)
TypeError: Cannot read property 'path' of null
    at Object.ID_1 (/Users/username/Sites/myapp/build/webpack:/src/stores/AppStore.js:69:26)
    at Dispatcher.$Dispatcher_invokeCallback (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:220:35)
    at Dispatcher.dispatch (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:195:14)
    at /Users/username/Sites/myapp/build/webpack:/src/core/Database.js:45:18
    at run (/Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:89:1)
    at /Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:100:1
    at process._tickCallback (node.js:355:11)
TypeError: Cannot read property 'path' of null
    at Object.ID_1 (/Users/username/Sites/myapp/build/webpack:/src/stores/AppStore.js:69:26)
    at Dispatcher.$Dispatcher_invokeCallback (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:220:35)
    at Dispatcher.dispatch (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:195:14)
    at /Users/username/Sites/myapp/build/webpack:/src/core/Database.js:45:18
    at run (/Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:89:1)
    at /Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:100:1
    at process._tickCallback (node.js:355:11)
TypeError: Cannot read property 'path' of null
    at Object.ID_1 (/Users/username/Sites/myapp/build/webpack:/src/stores/AppStore.js:69:26)
    at Dispatcher.$Dispatcher_invokeCallback (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:220:35)
    at Dispatcher.dispatch (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:195:14)
    at /Users/username/Sites/myapp/build/webpack:/src/core/Database.js:45:18
    at run (/Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:89:1)
    at /Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:100:1
    at process._tickCallback (node.js:355:11)
TypeError: Cannot read property 'path' of null
    at Object.ID_1 (/Users/username/Sites/myapp/build/webpack:/src/stores/AppStore.js:69:26)
    at Dispatcher.$Dispatcher_invokeCallback (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:220:35)
    at Dispatcher.dispatch (/Users/username/Sites/myapp/node_modules/flux/lib/Dispatcher.js:195:14)
    at /Users/username/Sites/myapp/build/webpack:/src/core/Database.js:45:18
    at run (/Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:89:1)
    at /Users/username/Sites/myapp/build/webpack:/~/babel-core/~/core-js/modules/es6.promise.js:100:1
    at process._tickCallback (node.js:355:11)myapp
```

**Comment by alem0lars** [comment]
I also have the same problem..

**Comment by staxmanade** [comment]
same here

**Comment by polytechnic** [comment]
Adding IF Statement in AppStore.js seems to fix the issue
'''
    case ActionTypes.RECEIVE_PAGE:
      loading = false;
      if (!action.err) {
        if (action.page) {
          pages[action.page.path] = action.page;
        }
      }
      AppStore.emitChange();
      break;
'''

**Comment by staxmanade** [comment]
@polytechnic, adding the `if...` avoids the server error, however it doesn't return a page just a `404`

`http://localhost:3000/api/query?path=/contact`

```
{
  "error": "The page '/contact' is not found."
}
```

**Comment by staxmanade** [comment]
Oh - I just created `/src/content/contact.jade` and it's returning the data...

**Comment by alem0lars** [comment]
I think the whole logic is wrong here.. We should ask the Database for a content only for ContactPage not every page.

**Comment by staxmanade** [comment]
@alem0lars - that's where I think I ended up - missing conceptually why we're asking the db for pages that are generated in other ways than from jade files... Hopefully @koistya can chime in and help fill in my gaps in understanding.

**Comment by alem0lars** [comment]
IMHO a project skeleton shouldn't go that in deep with having a custom manually implemented Database to load *some* pages. The first thing you're going to do in a real project is to remove that Database, the API calls..

Should that stuff still be in the skeleton or it should be just an example on how to (possibly) use it?

**Comment by devnomad** [comment]
@staxmanade it just seems that there's an API call to query the content for each page, regardless of its "type" (dynamic vs static).

The logic lays in *loadPage* (*src/actions/AppActions.js*). The quick fix would look like the one in 2944093c867977461c5b3cf52c6b1a6313e957a8 to prevent unnecessary calls (without forgetting the callback to maintain *navigateTo* history functionality).

I guess that routing (e.g. https://github.com/rackt/react-router) will solve this. Any thoughts @koistya ? Thanks for the good work btw :+1: !

**Comment by langpavel** [comment]
Quite old...
